src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm.cpp

```diff
@@ -153,31 +155,49 @@ std::shared_ptr<BrgemmCompiledKernel> BrgemmKernelExecutor::compile_kernel(const
 
     return compiled_kernel;
 }
-void BrgemmKernelExecutor::update_config(const ov::snippets::lowered::ExpressionPtr& expr, BrgemmKernelConfig& config) const {
-    auto get_projected_input_subtensor = [](const snippets::lowered::PortDescriptorPtr& desc) {
-        // Note: for output shape you will need get_preordered_vdims()
-        auto shape = snippets::utils::get_planar_vdims(desc->get_shape(), desc->get_layout());
-        auto subtensor = desc->get_subtensor();
-        OV_CPU_JIT_EMITTER_ASSERT(subtensor.size() <= shape.size() && subtensor.size() == 2,
-                                  "Invalid subtensor + shape combination");
-        auto shape_it = shape.rbegin();
-        for (auto sub_it = subtensor.rbegin(); sub_it != subtensor.rend(); sub_it++, shape_it++) {
-            *sub_it = std::min(*sub_it, *shape_it);
-        }
-        return subtensor;
-    };
+void BrgemmKernelExecutor::update_config(const ov::snippets::lowered::ExpressionPtr& expr,
+                                         const ov::snippets::lowered::LinearIRPtr& linear_ir,
+                                         BrgemmKernelConfig& config) const {
     const auto& input_pds = expr->get_input_port_descriptors();
     const auto& output_pds = expr->get_output_port_descriptors();
     OV_CPU_JIT_EMITTER_ASSERT((input_pds.size() == 2 || input_pds.size() == 3) && output_pds.size() == 1,
                               "Invalid number of in/out port descriptors");
-    // Update runtime-defined config fields:
-    // Matrix A (first input)
+
+    const auto in0_shape = snippets::utils::get_planar_vdims(input_pds[0]->get_shape(), input_pds[0]->get_layout());
+    const auto in1_shape = snippets::utils::get_planar_vdims(input_pds[1]->get_shape(), input_pds[1]->get_layout());
+    auto in0_subtensor = input_pds[0]->get_subtensor();
+    auto in1_subtensor = input_pds[1]->get_subtensor();
+
+    auto M = *++in0_subtensor.rbegin();
+    auto K = *in0_subtensor.rbegin();
+    auto N = *in1_subtensor.rbegin();
+
+    if (ov::snippets::utils::is_full_dim_value(M)) {
+        M = *++in0_shape.rbegin();
+    } else {
+        const auto& loop_ids = expr->get_loop_ids();
+        OPENVINO_ASSERT(!loop_ids.empty(), "Loop by dimension M is missed");
+        // TODO [146125]: Loop by M is first one in `loop_ids`
+        const auto& expanded_loop_info = linear_ir->get_loop_manager()->get_loop_info<ov::snippets::lowered::ExpandedLoopInfo>(loop_ids.front());
```

Comment:
It seems that we can easily check that the first loop is by M: 
1) get input ports => get expression port => check that it's 0th MatMul input (so it's either M or K)
2) get output ports => get_expression_ports => check that MatMul's output is incremented (must be M, since K loop does not increment output ptr).

Reply:
Also, how does it handle static case, when M < M_blk => so we don't have an M loop?

Reply:
Discussed offline.
Agreed to add loop validation in the next PR (Blocking)